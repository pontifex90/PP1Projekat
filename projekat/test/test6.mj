
program test6

int aa, bb;

{	
	int someMethod(int x, int y) {
		return x + y;
	}
	
	void printSomething(char c) {
		print(c);
	}
	
	void main()
	int ad;
	bool boolArray[];
	{
		boolArray = new bool[3];
		
		boolArray[ad + bb] = false;
		aa = someMethod(aa, bb);
		
		printSomething('c');
	}
}

