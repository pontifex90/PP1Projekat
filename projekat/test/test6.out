INFO  2017-02-11 06:19:17,627 [main][MJParser.java:615] : Compiling source file: D:\PP1ProjekatRepo\PP1Projekat\projekat\test\test6.mj
INFO  2017-02-11 06:19:17,639 [main][MJParser.java:588] : #2 program
INFO  2017-02-11 06:19:17,640 [main][MJParser.java:588] : #49 test6
INFO  2017-02-11 06:19:17,640 [main][MJParser.java:588] : #49 int
DEBUG 2017-02-11 06:19:17,641 [main][MJParser.java:770] : Semantic info on line 2: added program "test6" to symbol table. [ kind: 6  name: test6  type: 0]
INFO  2017-02-11 06:19:17,643 [main][MJParser.java:588] : #49 aa
INFO  2017-02-11 06:19:17,643 [main][MJParser.java:588] : #12 ,
DEBUG 2017-02-11 06:19:17,643 [main][MJParser.java:770] : Semantic info on line 4: added global variable "aa" to symbol table. [ kind: 1  name: aa  type: 1]
INFO  2017-02-11 06:19:17,643 [main][MJParser.java:588] : #49 bb
INFO  2017-02-11 06:19:17,643 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:17,644 [main][MJParser.java:770] : Semantic info on line 4: added global variable "bb" to symbol table. [ kind: 1  name: bb  type: 1]
INFO  2017-02-11 06:19:17,644 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:17,644 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:17,644 [main][MJParser.java:588] : #49 someMethod
INFO  2017-02-11 06:19:17,644 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:17,644 [main][MJParser.java:770] : Semantic info on line 7: added  method "someMethod" to symbol table. [ kind: 3  name: someMethod  type: 1]
INFO  2017-02-11 06:19:17,645 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:17,646 [main][MJParser.java:588] : #49 x
INFO  2017-02-11 06:19:17,646 [main][MJParser.java:588] : #12 ,
DEBUG 2017-02-11 06:19:17,646 [main][MJParser.java:770] : Semantic info on line 7: added formal parameter variable "x" to symbol table. [ kind: 1  name: x  type: 1]
INFO  2017-02-11 06:19:17,646 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:17,646 [main][MJParser.java:588] : #49 y
INFO  2017-02-11 06:19:17,646 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:19:17,646 [main][MJParser.java:770] : Semantic info on line 7: added formal parameter variable "y" to symbol table. [ kind: 1  name: y  type: 1]
INFO  2017-02-11 06:19:17,646 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:17,646 [main][MJParser.java:588] : #27 return
INFO  2017-02-11 06:19:17,647 [main][MJParser.java:588] : #49 x
INFO  2017-02-11 06:19:17,647 [main][MJParser.java:588] : #28 +
DEBUG 2017-02-11 06:19:17,647 [main][MJParser.java:770] : Usage info on line 8: formal parameter variable "x" used. [ kind:1  name: x  type: 1]
INFO  2017-02-11 06:19:17,647 [main][MJParser.java:588] : #49 y
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:17,648 [main][MJParser.java:770] : Usage info on line 8: formal parameter variable "y" used. [ kind:1  name: y  type: 1]
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #17 void
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #49 printSomething
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:17,648 [main][MJParser.java:770] : Semantic info on line 11: added  method "printSomething" to symbol table. [ kind: 3  name: printSomething  type: 0]
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #49 char
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #49 c
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:19:17,648 [main][MJParser.java:770] : Semantic info on line 11: added formal parameter variable "c" to symbol table. [ kind: 1  name: c  type: 2]
INFO  2017-02-11 06:19:17,648 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #23 print
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #9 (
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #49 c
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:19:17,649 [main][MJParser.java:770] : Usage info on line 12: formal parameter variable "c" used. [ kind:1  name: c  type: 2]
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #17 void
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #49 main
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:17,649 [main][MJParser.java:770] : Semantic info on line 15: added  method "main" to symbol table. [ kind: 3  name: main  type: 0]
INFO  2017-02-11 06:19:17,649 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #49 ad
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:17,650 [main][MJParser.java:770] : Semantic info on line 16: added local variable "ad" to symbol table. [ kind: 1  name: ad  type: 1]
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #49 bool
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #49 boolArray
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #13 [
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #14 ]
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:17,650 [main][MJParser.java:770] : Semantic info on line 17: added local array variable "boolArray" to symbol table. [ kind: 1  name: boolArray  type: 3]
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #49 boolArray
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:17,650 [main][MJParser.java:770] : Usage info on line 19: local variable "boolArray" used. [ kind:1  name: boolArray  type: 3]
INFO  2017-02-11 06:19:17,650 [main][MJParser.java:588] : #21 new
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #49 bool
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #13 [
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #48 3
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #14 ]
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:17,651 [main][MJParser.java:770] : Usage info on line 19: instantiated new object of type "array of bool".
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #49 boolArray
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:17,651 [main][MJParser.java:770] : Usage info on line 21: local variable "boolArray" used. [ kind:1  name: boolArray  type: 3]
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #49 ad
INFO  2017-02-11 06:19:17,651 [main][MJParser.java:588] : #28 +
DEBUG 2017-02-11 06:19:17,651 [main][MJParser.java:770] : Usage info on line 21: local variable "ad" used. [ kind:1  name: ad  type: 1]
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #49 bb
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #14 ]
DEBUG 2017-02-11 06:19:17,652 [main][MJParser.java:770] : Usage info on line 21: global variable "bb" used. [ kind:1  name: bb  type: 1]
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:17,652 [main][MJParser.java:770] : Usage info on line 21: element of array "boolArray" used. [ kind:1  name: boolArray  type: 3]
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #51 false
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #49 aa
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:17,652 [main][MJParser.java:770] : Usage info on line 22: global variable "aa" used. [ kind:1  name: aa  type: 1]
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #49 someMethod
INFO  2017-02-11 06:19:17,652 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:17,652 [main][MJParser.java:770] : Usage info on line 22: global method "someMethod" used. [ kind:3  name: someMethod  type: 1]
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #49 aa
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #12 ,
DEBUG 2017-02-11 06:19:17,653 [main][MJParser.java:770] : Usage info on line 22: global variable "aa" used. [ kind:1  name: aa  type: 1]
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #49 bb
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:19:17,653 [main][MJParser.java:770] : Usage info on line 22: global variable "bb" used. [ kind:1  name: bb  type: 1]
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #49 printSomething
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:17,653 [main][MJParser.java:770] : Usage info on line 24: global method "printSomething" used. [ kind:3  name: printSomething  type: 0]
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #50 c
INFO  2017-02-11 06:19:17,653 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:17,654 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:17,654 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:17,654 [main][MJParser.java:588] : #8 }
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog test6: notype, 0, 1 
   Var aa: int, 0, 0 
   Var bb: int, 1, 0 
   Meth someMethod: int, 0, 2 
      Var x: int, 0, 1 
      Var y: int, 1, 1 

   Meth printSomething: notype, 8, 1 
      Var c: char, 0, 1 

   Meth main: notype, 16, 0 
      Var ad: int, 0, 1 
      Var boolArray: Arr of , 1, 1 



INFO  2017-02-11 06:19:17,659 [main][MJParser.java:633] : Parsing finished successfully!
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:636] : 
===============================
VARS COUNTING:
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:637] : Constants number: 0
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:638] : Global vars number: 2
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:639] : Global array vars number: 0
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:640] : Class vars number: 0
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:641] : Main function vars number: 2
===============================

INFO  2017-02-11 06:19:17,660 [main][MJParser.java:642] : 
===============================
FUNCTION CALLS COUNTING:
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:643] : Global functions: 3
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:644] : Class functions: 0
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:645] : Class static functions: 0
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:646] : Statements block number: 0
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:647] : Main method function calls number: 1
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:648] : Function formal parameters number: 3
INFO  2017-02-11 06:19:17,660 [main][MJParser.java:649] : Class definitions number: 0
